accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,199 @@
+
+## Peripheral software channels creation
+
+We need a mechanism to create the channels in the peripheral servers (vendor and CLM's) in the desired organization. The peripheral channel creation must be done from the HUB server through an API. Since we are making special channel creation (defined next), those API methods should be available to server-to-server communication only.
+
+For each peripheral we should define the organization mapping between HUB organizations and peripheral ones, following the rules (similiar to what we have on ISSv1):
+  - Each Hub organization can only be mapped to one peripheral organizations
+  - Peripheral organization can receive data from several different HUB organizations
+
+From the webUI we should be able to select which channels should be added to each peripheral (vendor or custom). This needed to be saved in a new database table on the HUB side.
+
+Steps needed to create the channels on peripheral:
+   - In the HUB, generate authentication tokens to access the channel(s). Re-use the existing channel access table.
+   - In the peripheral create all needed channels (vendor and custom) point to the HUB server in the rhncontentsource table. Each channel needs to have its own entry.
+   - Minimal set of tables will be populated on the peripheral: rhnchannel,  rhnchannelcontentsource, rhnchannelproduct, rhnchannelfamilymembers, rhnproductname, rhnchannelcloned, suseproductchannel.
+
+
+When creating the channels in the peripheral server the content source will reference the correct HUB repository that provides that channel. Peripheral servers need to be configured with the flag `java.unify_custom_channel_management`, which will synchronize custom channels during the nightly mgr-sync process.
+
+One important aspect is to recreate the connection between custom channels and vendor channels, so we can have SP migration and avoid the need to synchronize all channel clone chains (ISSv1 also does this implementation).
+
+Vendor channels are not linked to any organization, and can also be synchronized with this method.
+
+## Communication Workflow
+
+![Communication workflow](images/hub_sync_content/comunication_workflow.png)
+
+## Integration with SUSE Manager 5.0
+
+With SUSE Manager containerization we cannot register the peripheral servers as minions anymore. Several components were relying on having that.
+In this section we will have a walk-through of those problems and how this solution can help solve them.
+
+### HUB XML-RPC API
+
+Component responsible for broadcasting API requests to peripheral servers. This component loads the peripheral servers that are registered on the HUB by looking at the minion entitlements.
+
+With the proposed solution we can retrieve the list of the registered peripheral servers and avoid the peripheral server registration as minion.
+
+### Inter-server-sync-v2
+
+Will be replaced by this solution. It will continue to exist but for disconnect set-ups only.
```

Comment:
May be this should also state that ISSv2 will also stay for image and config channels sync until this is implemented in the new solution.

Reply:
This is mentioned in the tools scope: https://github.com/uyuni-project/uyuni-rfc/pull/82/files#diff-2cf517457bdce7fb7b50dd31c7c74dcee549a8e64965e96221fac04f05cfe5a0R90

Reply:
If stating it once and for all is fine, then I'm good with this

Reply:
In the mean time I added that information to this sentence. It would not hurt and bring claririty.